shared/src/main/scala/mlscript/TyperDatatypes.scala

```diff
@@ -31,34 +35,30 @@ abstract class TyperDatatypes extends TyperHelpers { self: Typer =>
     def rigidify(implicit lvl: Int): SimpleType = freshenAbove(level, body, rigidify = true)
   }
   
-  class MethodType(level: Int, body: SimpleType, val prts: List[TypeName], val single: Bool)
-      extends PolymorphicType(level, body) {
+  class MethodType(val level: Int, val body: Opt[SimpleType], val parents: List[TypeName], val single: Bool)
```

Comment:
You could make this `case class MethodType(level: Int, body: Opt[SimpleType], parents: List[TypeName], single: Bool)` and avoid having to implement `copy`, `apply`, and `unapply`. Also, could you document what `single` is for?

Reply:
I wrote it this way to avoid explicitly dealing with `single` outside the class, which can only be turned into `false` by `&`. The documentation is added in dbf9e69d1f0b3293e36d97c6cf378bdfdb422fa8

Reply:
Ok sure. Not sure it's worth it, but it's a detail.